spec/ParseAPI.spec.js

```diff
@@ -429,6 +429,47 @@ describe('miscellaneous', function() {
     });
   });
 
+  it('test beforeSave get dirtyKeys on update', function(done) {
+    var triggerTime = 0;
+    // Register a mock beforeSave hook
+    Parse.Cloud.beforeSave('GameScore', function(req, res) {
+      var object = req.object;
+      expect(object instanceof Parse.Object).toBeTruthy();
+      expect(object.get('fooAgain')).toEqual('barAgain');
+      if (triggerTime == 0) {
+        // Create
+        expect(object.get('foo')).toEqual('bar');
+      } else if (triggerTime == 1) {
+        // Update
+        expect(object.dirtyKeys()).toEqual(['foo']);
+        expect(object.dirty('foo')).toBeTruthy();
+        expect(object.get('foo')).toEqual('baz');
+      } else {
+        res.error();
+      }
+      triggerTime++;
+      res.success();
+    });
+
+    var obj = new Parse.Object('GameScore');
+    obj.set('foo', 'bar');
+    obj.set('fooAgain', 'barAgain');
+    obj.save().then(function() {
+      // We only update foo
+      obj.set('foo', 'baz');
+      return obj.save();
+    }).then(function() {
+      // Make sure the checking has been triggered
+      expect(triggerTime).toBe(2);
+      // Clear mock beforeSave
+      delete Parse.Cloud.Triggers.beforeSave.GameScore;
```

Comment:
They actually do  one idea might be to randomize the class name - so it's different every time, but is the same for the entire function.
Oh... Also, I believe we need a huge afterTest here, since if the test fails - other tests are pretty much doomed.